app/assets/javascripts/components/tabs.js

```diff
@@ -88,12 +88,24 @@
     }
   });
 
-  // Show the first tabPanel
-  $tabs.find('.tabPanel:first').attr('aria-hidden', 'false').show();
-
-  // Set state for the first tabsList li
-  $tabsList.find('li:first').addClass('current').find(' > a').attr({
-    'aria-selected': true,
-    tabindex: 0
-  });
+  var $anchor_current;
+  var stripped_url = document.location.toString().split("#");
+  if (stripped_url.length > 1) {
+    // Does an anchor with the correct id exist with this anchor value?
+    $anchor_current = $('[id="tab-' + stripped_url[1] + '"]:first');
+  }
+
+  if ($anchor_current && $anchor_current.length == 1) {
+    // Show the tabPanel specified by the anchor
+    $anchor_current.trigger('click');
+  } else {
+    // Show the first tabPanel
+    $tabs.find('.tabPanel:first').attr('aria-hidden', 'false').show();
+
+    // Set state for the first tabsList li
+    $tabsList.find('li:first').addClass('current').find(' > a').attr({
+      'aria-selected': true,
+      tabindex: 0
+    });
+  }
```

Comment:
Would it be possible to rework this conditional logic in a way that behaves more like…

1. Test to see if there's a fragment identifier in the URL,
2. If so, grab that string and jQuery references to the appropriate tab and panel,
3. If not, grab a reference to the first tab and tab panel,
4. Do the necessary DOM manipulation to initialize the correct tab and panel.

The variable names in 2 and 3 would be the same so that 4 would be a generic initialization.